gramps/gen/lib/note.py

```diff
@@ -272,6 +273,18 @@ def get(self):
         """
         return str(self.text)
 
+    def get_preview(self):
+        """Return the max visible text string associated with the note.
+
+        :returns: either the *clear* text of the note contents or the maximum
+                  visible text.
+        :rtype: unicode
+        """
+        max_vis_len = config.get("interface.note-max-vis-length")
+        if len(str(self.text)) > max_vis_len:
+            return str(self.text).replace("\n", " ")[:max_vis_len] + "..."
+        return str(self.text)
```

Comment:
You will still need to replace the "\n" with " " here.

Reply:
Yes we need that because get_preview is only used to be displayed in GUI. If we don't do that, we can have multiple lines (i.e in the noteview like below)
![multiple-lines](https://github.com/user-attachments/assets/be32cbc2-3bb0-4f77-b382-1aa8beb2fd0a)

Reply:
Were you talking about the second return?

Reply:
Yes.

Reply:
Done